app/controllers/resources_controller.rb

```diff
@@ -1,12 +1,23 @@
 # frozen_string_literal: true
 
+require 'page_util'
+
 class ResourcesController < ApplicationController
   def index
     render json: all_resources, include: params[:include], status: :ok
   end
 
   def show
-    resource
+    render json: load_resource, include: params[:include], status: :ok
+  end
+
+  def update
+    a = authorize!
```

Comment:
This should probably be implemented using a filter. It'll allow you to enforce authorized! before this action is called.
See http://guides.rubyonrails.org/action_controller_overview.html#filters
Something like
```ruby
before_action :authorize!, only: :show
```
You can actually add it to a base controller to enforce all actions, the use `skip_before_action` in specific controllers to skip authorization if it isn't needed.